http://bugs.gentoo.org/401851
http://sourceforge.net/tracker/?func=detail&aid=3504801&group_id=15273&atid=115273

the x86 asm code is not PIC friendly, so disable it and fall back to the
gcc builtin for doing prefetching

--- src/lists.h
+++ src/lists.h
@@ -68,7 +68,7 @@
     asm volatile ("lfetch [%0]"    :: "r" (x))
 #elif defined(__powerpc64__)
     asm volatile ("dcbt 0,%0"      :: "r" (x))
-#elif !defined(__CYGWIN__) && defined(__i386__)
+#elif !defined(__CYGWIN__) && !defined(__PIC__) && defined(__i386__)
     asm volatile ("661:\n\t"
 		  ".byte 0x8d,0x74,0x26,0x00\n"
 		  "\n662:\n"
